Sources/NIOHTTP1Server/main.swift

```diff
@@ -33,6 +33,18 @@ extension String {
     }
 }
 
+private func httpResponseHead(request: HTTPRequestHead, status: HTTPResponseStatus, headers: HTTPHeaders = HTTPHeaders()) -> HTTPResponseHead {
+    var head = HTTPResponseHead(version: request.version, status: status, headers: headers)
+    if request.isKeepAlive && request.version.major == 1 && request.version.minor == 0 {
+        /* to work around `ab -k` behaving weirdly: it'll send an HTTP/1.0 request with keep-alive set
+           but stops doing anything at all if the server doesn't also set Connection: keep-alive */
+        if head.headers["connection"].isEmpty {
```

Comment:
@weissi what I was saying was the the user may have already added a `connection` header with another value then `keep-alive` so you need to check if its easier empty or the values don't include "keep-alive".

Reply:
@weissi maybe I am just "to sensitive" here but in theory a user could have a connection header which is neither `close` nor `keep-alive`. In this case you still would like to add `keep-alive` if I am not wrong.

Reply:
You're not too sensitive @normanmaurer, that's definitely right.

`Connection` should contain, in addition to the flag indicating connection behaviour, any header that is explicitly hop-by-hop (and that should be stripped by intervening HTTP middleboxes like proxy). We can't blindly assume that a user has set either keep-alive or close: we do need to check.

Reply:
not sure if quietly sanitising that is fine but let's do it. @normanmaurer / @Lukasa let me know what you think about this.